python/cudf/cudf/pandas/_wrappers/pandas.py

```diff
@@ -1776,6 +1781,65 @@ def wrapped_call(cls, data, *args, **kwargs):
     return wrapped_call
 
 
+def wrap_from_pandas(original_call):
+    @functools.wraps(original_call)
+    def wrapped_from_pandas(obj, *args, **kwargs):
+        if is_proxy_object(obj):
+            obj = obj.as_gpu_object()
+            return obj
+        return original_call(obj, *args, **kwargs)
+
+    return wrapped_from_pandas
+
+
+def wrap_from_pandas_dataframe(original_call):
+    @functools.wraps(original_call)
+    def wrapped_from_pandas_dataframe(dataframe, *args, **kwargs):
+        if is_proxy_object(dataframe):
+            dataframe = dataframe.as_gpu_object()
+            if isinstance(dataframe, cudf.DataFrame):
+                return dataframe
+        return original_call(dataframe, *args, **kwargs)
+
+    return wrapped_from_pandas_dataframe
+
+
+def wrap_from_pandas_series(original_call):
+    @functools.wraps(original_call)
+    def wrapped_from_pandas_series(s, *args, **kwargs):
+        if is_proxy_object(s):
+            s = s.as_gpu_object()
+            if isinstance(s, cudf.Series):
+                return s
+        return original_call(s, *args, **kwargs)
+
+    return wrapped_from_pandas_series
+
+
+def wrap_from_pandas_index(original_call):
+    @functools.wraps(original_call)
+    def wrapped_from_pandas_index(index, *args, **kwargs):
+        if is_proxy_object(index):
+            index = index.as_gpu_object()
+            if isinstance(index, cudf.core.index.BaseIndex):
+                return index
+        return original_call(index, *args, **kwargs)
+
+    return wrapped_from_pandas_index
+
+
+def wrap_from_pandas_multiindex(original_call):
+    @functools.wraps(original_call)
+    def wrapped_from_pandas_multiindex(multiindex, *args, **kwargs):
+        if is_proxy_object(multiindex):
+            multiindex = multiindex.as_gpu_object()
+            if isinstance(multiindex, cudf.MultiIndex):
+                return multiindex
+        return original_call(multiindex, *args, **kwargs)
+
+    return wrapped_from_pandas_multiindex
```

Comment:
Could this work?
```python
def wrap_from_pandas(original_call, type_=None):
    @functools.wraps(original_call)
    def wrapped(obj, *args, **kwargs)
        if is_proxy_object(obj):
            obj = obj.as_gpu_object()
            if (type_ is None) or isinstance(obj, type_):
                return obj
        return original_call(obj, *args, **kwargs)
    return wrapped
```

Reply:
The problem will arise if some passed keywords like `cudf.DataFrame.from_pandas(dataframe=df)` , `cudf.Series.from_pandas(s=ser)`. Your suggestion was the way I initially wanted to go by but all the functions have different keyword argument names so I had to separate out the implementations.

Reply:
We could presumably handle that in `kwargs` in the unified wrapper? This isn't a blocker btw

Reply:
That approach might get very complicated with a bunch of if/elif's to handle all possible combinations of each constructor.